src/wasm-linker.cpp

```diff
@@ -387,10 +389,29 @@ bool hasI64ResultOrParam(FunctionType* ft) {
   return false;
 }
 
+void Linker::makeDummyFunction() {
+  bool create = false;
+  // Check if there are address-taken functions
+  for (auto& relocation : out.relocations) {
+    if (relocation->kind == LinkerObject::Relocation::kFunction) {
+      create = true;
+      break;
+    }
+  }
+  if (!create || out.wasm.table.names.size()) return;
```

Comment:
This check means that if there's anything in the table, we bail, which I guess is for the case where the indexes are pre-assigned. Otherwise if that happens, we are basically already in a screwed-up state, right? Can we put an assert before we run the pre-assignment that the table is empty?